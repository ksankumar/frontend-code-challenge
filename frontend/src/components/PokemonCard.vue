<template>
  <v-card
    outlined
    @click.stop="goPokemonDetails(pokemon.id)"
    class="pokemon-card"
  >
    <v-img :src="pokemon.image" class="text-right pa-2" contain height="300">
      <template v-slot:placeholder>
        <v-skeleton-loader v-slot:placeholder type="image"></v-skeleton-loader>
      </template>
    </v-img>
    <v-card-actions class="grey lighten-3">
      <div>
        <div class="font-weight-normal">
          <strong>{{ pokemon.name }}</strong>
        </div>
        <div v-if="pokemon.types">{{ pokemon.types.join(", ") }}</div>
      </div>
      <v-spacer></v-spacer>
      <favorite-button :pokemon="pokemon" />
    </v-card-actions>
  </v-card>
</template>

<script>
import FavoriteButton from "@/components/FavoriteButton";

export default {
  name: "pokemon-card",
  components: {
    FavoriteButton
  },
  props: {
    pokemon: {
      type: Object
    }
  },
  methods: {
    goPokemonDetails(pokemonId) {
      this.$router.push({
        name: "PokemonDetails",
        params: {
          id: pokemonId
        }
      });
    }
  }
};
</script>
<style scoped>
@-webkit-keyframes swing {
  15% {
    -webkit-transform: translateX(5px);
    transform: translateX(5px);
  }
  30% {
    -webkit-transform: translateX(-5px);
    transform: translateX(-5px);
  }
  50% {
    -webkit-transform: translateX(3px);
    transform: translateX(3px);
  }
  65% {
    -webkit-transform: translateX(-3px);
    transform: translateX(-3px);
  }
  80% {
    -webkit-transform: translateX(2px);
    transform: translateX(2px);
  }
  100% {
    -webkit-transform: translateX(0);
    transform: translateX(0);
  }
}
@keyframes swing {
  15% {
    -webkit-transform: translateX(5px);
    transform: translateX(5px);
  }
  30% {
    -webkit-transform: translateX(-5px);
    transform: translateX(-5px);
  }
  50% {
    -webkit-transform: translateX(3px);
    transform: translateX(3px);
  }
  65% {
    -webkit-transform: translateX(-3px);
    transform: translateX(-3px);
  }
  80% {
    -webkit-transform: translateX(2px);
    transform: translateX(2px);
  }
  100% {
    -webkit-transform: translateX(0);
    transform: translateX(0);
  }
}
.pokemon-card:hover {
  -webkit-animation: swing 1s ease;
  animation: swing 1s ease;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
}
</style>
