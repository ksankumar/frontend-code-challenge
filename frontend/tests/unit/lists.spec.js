import { shallowMount, createLocalVue } from "@vue/test-utils";
import { createMockClient } from "mock-apollo-client";
import Vue from "vue";
import Vuetify from "vuetify";
import VueRouter from "vue-router";
import VueApollo from "vue-apollo";
import mockPokemonList from "../data/list.json";
import mockPokemonTypes from "../data/types.json";
import pokemonsQuery from "@/graphql/pokemons.query.gql";
import pokemonTypesQuery from "@/graphql/pokemonTypes.query.gql";
import localTestQuery from "@/graphql/localTest.query.gql";
import PokemonsComponent from "@/views/List";
import PokemonCard from "@/components/PokemonCard";
import NoDataComponent from "@/components/NoData";

const localVue = createLocalVue();

// let wrapper
const routes = [
  { path: "/", name: "PokemonsList", component: PokemonsComponent }
];
const router = new VueRouter({ routes });

Vue.use(VueRouter);
Vue.use(Vuetify);
Vue.use(VueApollo);

describe("Pokemons List view", () => {
  let wrapper;
  let mockClient;
  let apolloProvider;
  let requestHandlers;

  const createComponent = (handlers, data) => {
    // We want to specify default handlers for different operations and overwrite them with passed handlers
    requestHandlers = {
      allPokeyMonsQueryHandler: jest.fn().mockResolvedValue(mockPokemonList),
      pokeyMonsTypesQueryHandler: jest.fn().mockResolvedValue(mockPokemonTypes),
      // allPokeyMonsQueryHandler: jest.fn().mockResolvedValue(mockPokemonList),
      ...handlers
    };

    mockClient = createMockClient({
      resolvers: {}
    });

    mockClient.cache.writeQuery({
      query: localTestQuery,
      data: {
        fetchLocalUser: {
          __typename: "Pokemon",
          name: "Test"
        }
      }
    });

    // Unfortunately, it's not possible to override handler already set
    // So we need to do all the work for setting handlers in the factory
    mockClient.setRequestHandler(
      pokemonsQuery,
      requestHandlers.allPokeyMonsQueryHandler
    );
    mockClient.setRequestHandler(
      pokemonTypesQuery,
      requestHandlers.pokeyMonsTypesQueryHandler
    );

    apolloProvider = new VueApollo({
      defaultClient: mockClient
    });
    wrapper = shallowMount(PokemonsComponent, {
      localVue,
      apolloProvider,
      router,
      data() {
        return {
          ...data
        };
      }
    });
  };

  afterEach(() => {
    wrapper.destroy();
    mockClient = null;
    apolloProvider = null;
  });
  // it("add a pokemon to favorite on Favorite mutation", async () => {
  //   createComponent();
  //   await wrapper.vm.$nextTick();
  //   await wrapper.vm.$nextTick();
  //   // Waiting for query promise to resolve and populate data
  //   try {
  //     wrapper
  //       .findAllComponents(PokemonCard)
  //       .at(0)
  //       .vm.makeFavorite({ id: "001", isFavorite: true });
  //     wrapper
  //       .findAllComponents(PokemonCard)
  //       .at(0)
  //       .vm.$emit("makeFavorite", { id: "001", isFavorite: true });
  //   } catch (e) {
  //     console.log(e);
  //   }
  //
  //   // expect(requestHandlers.deleteHeroMutationHandler).toHaveBeenCalledWith({
  //   //   name: "Anonymous Vue Hero"
  //   // });
  //   // await wrapper.vm.$nextTick();
  //   // await wrapper.vm.$nextTick();
  //   // wrapper
  //   //   .findAllComponents(PokemonCard)
  //   //   .at(0)
  //   //   .vm.$emit("makeFavorit1e", { isFavorite: true });
  //
  //   // expect(requestHandlers.addHeroMutationHandler).toHaveBeenCalledWith({
  //   //   hero: {
  //   //     ...newHeroMock
  //   //   }
  //   // });
  //
  //   // We wait for mutation promise to resolve and then we check if new hero is added to cache
  //   // await wrapper.vm.$nextTick();
  //   //
  //   // expect(
  //   //   mockClient.cache.readQuery({ query: allHeroesQuery }).allHeroes
  //   // ).toHaveLength(3);
  //   //
  //   // // We wait for one more tick for component to re-render updated cache data
  //   // await wrapper.vm.$nextTick();
  //   //
  //   // expect(wrapper.html()).toMatchSnapshot();
  //   // expect(wrapper.findAllComponents(VueHero)).toHaveLength(3);
  // });
  it("renders a loading block when query is in progress", async () => {
    createComponent();
    expect(requestHandlers.allPokeyMonsQueryHandler).toHaveBeenCalled();
    expect(wrapper.find(".test-loading").exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders a list of 10 pokemons when query is resolved", async () => {
    createComponent();
    await wrapper.vm.$nextTick();
    await wrapper.vm.$nextTick();

    expect(wrapper.find(".test-loading").exists()).toBe(false);
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.findAllComponents(PokemonCard)).toHaveLength(10);
  });

  it("renders a list of pokemon types when query is resolved", async () => {
    createComponent({
      pokeyMonsTypesQueryHandler: jest.fn().mockResolvedValue(mockPokemonTypes)
    });
    // For some reason, when we reject the promise, it requires +1 tick to render an error
    await wrapper.vm.$nextTick();
    await wrapper.vm.$nextTick();
    expect(requestHandlers.pokeyMonsTypesQueryHandler).toHaveBeenCalled();
    // expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders a message about no pokemons when heroes list is empty", async () => {
    createComponent({
      allPokeyMonsQueryHandler: jest
        .fn()
        .mockResolvedValue({ data: { pokemons: { edges: [] } } })
    });
    await wrapper.vm.$nextTick();
    await wrapper.vm.$nextTick();

    expect(wrapper.find(".test-loading").exists()).toBe(false);
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.findAllComponents(NoDataComponent)).toHaveLength(1);
  });

  it("renders error if query fails", async () => {
    createComponent({
      allPokeyMonsQueryHandler: jest
        .fn()
        .mockRejectedValue(new Error("GraphQL error"))
    });
    // For some reason, when we reject the promise, it requires +1 tick to render an error
    await wrapper.vm.$nextTick();
    await wrapper.vm.$nextTick();
    await wrapper.vm.$nextTick();

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.findAllComponents(NoDataComponent)).toHaveLength(1);
  });
});
