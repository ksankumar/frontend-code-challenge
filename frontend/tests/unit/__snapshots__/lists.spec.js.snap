// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemons List view renders a list of 10 pokemons when query is resolved 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" width="100%" flat="true">
  <div class="overflow-hidden">
    <v-row-stub tag="div" nogutters="true">
      <v-col-stub cols="6" sm="6" md="6" tag="div">
        <v-btn-stub color="primary" elevation="0" tile="true" tag="button" activeclass="" large="true" block="true" type="button" class="text-capitalize">
          All
        </v-btn-stub>
      </v-col-stub>
      <v-col-stub cols="6" sm="6" md="6" tag="div">
        <v-btn-stub color="primary" elevation="0" tile="true" outlined="true" tag="button" activeclass="" large="true" block="true" type="button" class="text-capitalize">
          Favorites
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" align="center" justify="space-around">
      <v-col-stub cols="12" sm="6" md="7" tag="div" class="pr-3">
        <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" hidedetails="true" label="Search" loaderheight="2" clearable="true" clearicon="$clear" outlined="true" type="text"></v-text-field-stub>
      </v-col-stub>
      <v-col-stub cols="8" sm="4" md="4" tag="div">
        <v-select-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" appendicon="$dropdown" backgroundcolor="" hidedetails="true" label="Type" loaderheight="2" clearable="true" clearicon="$clear" outlined="true" type="text" valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date &amp;&amp; b instanceof Date &amp;&amp; a.getTime() !== b.getTime()) {
    // If the values are Date, compare them as timestamps
    return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}" nodatatext="$vuetify.noDataText" items="Grass,Poison,Fire,Flying,Water,Bug,Normal,Electric,Ground,Fairy,Fighting,Psychic,Rock,Steel,Ice,Ghost,Dragon" itemcolor="primary" itemdisabled="disabled" itemtext="text" itemvalue="value" menuprops="[object Object]" class="selectable"></v-select-stub>
      </v-col-stub>
      <v-col-stub cols="4" sm="2" md="1" tag="div">
        <v-row-stub tag="div" nogutters="true" align="center" justify="space-around">
          <v-btn-stub color="primary" tag="button" activeclass="" large="true" icon="true" type="button">
            <v-icon-stub large="">mdi-view-list</v-icon-stub>
          </v-btn-stub>
          <v-divider-stub vertical="true" align="center"></v-divider-stub>
          <v-btn-stub color="primary" tag="button" activeclass="" large="true" icon="true" type="button">
            <v-icon-stub large="">mdi-grid</v-icon-stub>
          </v-btn-stub>
        </v-row-stub>
      </v-col-stub>
    </v-row-stub>
  </div>
  <v-divider-stub class="elevation-10"></v-divider-stub>
  <v-card-stub loaderheight="4" tag="div" flat="true" class="content-scroll">
    <v-slide-x-transition-stub mode="out-in" origin="top center 0" duration="250">
      <v-row-stub tag="div">
        <v-col-stub cols="12" sm="4" md="3" tag="div" class="pullUp">
          <pokemoncard-stub pokemon="[object Object]"></pokemoncard-stub>
        </v-col-stub>
        <v-col-stub cols="12" sm="4" md="3" tag="div" class="pullUp">
          <pokemoncard-stub pokemon="[object Object]"></pokemoncard-stub>
        </v-col-stub>
        <v-col-stub cols="12" sm="4" md="3" tag="div" class="pullUp">
          <pokemoncard-stub pokemon="[object Object]"></pokemoncard-stub>
        </v-col-stub>
        <v-col-stub cols="12" sm="4" md="3" tag="div" class="pullUp">
          <pokemoncard-stub pokemon="[object Object]"></pokemoncard-stub>
        </v-col-stub>
        <v-col-stub cols="12" sm="4" md="3" tag="div" class="pullUp">
          <pokemoncard-stub pokemon="[object Object]"></pokemoncard-stub>
        </v-col-stub>
        <v-col-stub cols="12" sm="4" md="3" tag="div" class="pullUp">
          <pokemoncard-stub pokemon="[object Object]"></pokemoncard-stub>
        </v-col-stub>
        <v-col-stub cols="12" sm="4" md="3" tag="div" class="pullUp">
          <pokemoncard-stub pokemon="[object Object]"></pokemoncard-stub>
        </v-col-stub>
        <v-col-stub cols="12" sm="4" md="3" tag="div" class="pullUp">
          <pokemoncard-stub pokemon="[object Object]"></pokemoncard-stub>
        </v-col-stub>
        <v-col-stub cols="12" sm="4" md="3" tag="div" class="pullUp">
          <pokemoncard-stub pokemon="[object Object]"></pokemoncard-stub>
        </v-col-stub>
        <v-col-stub cols="12" sm="4" md="3" tag="div" class="pullUp">
          <pokemoncard-stub pokemon="[object Object]"></pokemoncard-stub>
        </v-col-stub>
        <v-col-stub cols="12" sm="4" md="3" tag="div" class="pullUp">
          <pokemoncard-stub pokemon="[object Object]"></pokemoncard-stub>
        </v-col-stub>
        <v-col-stub cols="12" sm="4" md="3" tag="div" class="pullUp">
          <pokemoncard-stub pokemon="[object Object]"></pokemoncard-stub>
        </v-col-stub>
      </v-row-stub>
    </v-slide-x-transition-stub>
    <!---->
  </v-card-stub>
</v-card-stub>
`;

exports[`Pokemons List view renders a loading block when query is in progress 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" width="100%" flat="true">
  <div class="overflow-hidden">
    <v-row-stub tag="div" nogutters="true">
      <v-col-stub cols="6" sm="6" md="6" tag="div">
        <v-btn-stub color="primary" elevation="0" tile="true" tag="button" activeclass="" large="true" block="true" type="button" class="text-capitalize">
          All
        </v-btn-stub>
      </v-col-stub>
      <v-col-stub cols="6" sm="6" md="6" tag="div">
        <v-btn-stub color="primary" elevation="0" tile="true" outlined="true" tag="button" activeclass="" large="true" block="true" type="button" class="text-capitalize">
          Favorites
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" align="center" justify="space-around">
      <v-col-stub cols="12" sm="6" md="7" tag="div" class="pr-3">
        <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" hidedetails="true" label="Search" loaderheight="2" clearable="true" clearicon="$clear" outlined="true" type="text"></v-text-field-stub>
      </v-col-stub>
      <v-col-stub cols="8" sm="4" md="4" tag="div">
        <v-skeleton-loader-stub type="list-item" types="[object Object]"></v-skeleton-loader-stub>
      </v-col-stub>
      <v-col-stub cols="4" sm="2" md="1" tag="div">
        <v-row-stub tag="div" nogutters="true" align="center" justify="space-around">
          <v-btn-stub color="primary" tag="button" activeclass="" large="true" icon="true" type="button">
            <v-icon-stub large="">mdi-view-list</v-icon-stub>
          </v-btn-stub>
          <v-divider-stub vertical="true" align="center"></v-divider-stub>
          <v-btn-stub color="primary" tag="button" activeclass="" large="true" icon="true" type="button">
            <v-icon-stub large="">mdi-grid</v-icon-stub>
          </v-btn-stub>
        </v-row-stub>
      </v-col-stub>
    </v-row-stub>
  </div>
  <v-divider-stub class="elevation-10"></v-divider-stub>
  <v-card-stub loaderheight="4" tag="div" flat="true" class="content-scroll">
    <div class="test-loading">
      <v-skeleton-loader-stub type="image, list-item-avatar-two-line" types="[object Object]"></v-skeleton-loader-stub>
    </div>
    <!---->
  </v-card-stub>
</v-card-stub>
`;

exports[`Pokemons List view renders a message about no pokemons when heroes list is empty 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" width="100%" flat="true">
  <div class="overflow-hidden">
    <v-row-stub tag="div" nogutters="true">
      <v-col-stub cols="6" sm="6" md="6" tag="div">
        <v-btn-stub color="primary" elevation="0" tile="true" tag="button" activeclass="" large="true" block="true" type="button" class="text-capitalize">
          All
        </v-btn-stub>
      </v-col-stub>
      <v-col-stub cols="6" sm="6" md="6" tag="div">
        <v-btn-stub color="primary" elevation="0" tile="true" outlined="true" tag="button" activeclass="" large="true" block="true" type="button" class="text-capitalize">
          Favorites
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" align="center" justify="space-around">
      <v-col-stub cols="12" sm="6" md="7" tag="div" class="pr-3">
        <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" hidedetails="true" label="Search" loaderheight="2" clearable="true" clearicon="$clear" outlined="true" type="text"></v-text-field-stub>
      </v-col-stub>
      <v-col-stub cols="8" sm="4" md="4" tag="div">
        <v-select-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" appendicon="$dropdown" backgroundcolor="" hidedetails="true" label="Type" loaderheight="2" clearable="true" clearicon="$clear" outlined="true" type="text" valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date &amp;&amp; b instanceof Date &amp;&amp; a.getTime() !== b.getTime()) {
    // If the values are Date, compare them as timestamps
    return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}" nodatatext="$vuetify.noDataText" items="Grass,Poison,Fire,Flying,Water,Bug,Normal,Electric,Ground,Fairy,Fighting,Psychic,Rock,Steel,Ice,Ghost,Dragon" itemcolor="primary" itemdisabled="disabled" itemtext="text" itemvalue="value" menuprops="[object Object]" class="selectable"></v-select-stub>
      </v-col-stub>
      <v-col-stub cols="4" sm="2" md="1" tag="div">
        <v-row-stub tag="div" nogutters="true" align="center" justify="space-around">
          <v-btn-stub color="primary" tag="button" activeclass="" large="true" icon="true" type="button">
            <v-icon-stub large="">mdi-view-list</v-icon-stub>
          </v-btn-stub>
          <v-divider-stub vertical="true" align="center"></v-divider-stub>
          <v-btn-stub color="primary" tag="button" activeclass="" large="true" icon="true" type="button">
            <v-icon-stub large="">mdi-grid</v-icon-stub>
          </v-btn-stub>
        </v-row-stub>
      </v-col-stub>
    </v-row-stub>
  </div>
  <v-divider-stub class="elevation-10"></v-divider-stub>
  <v-card-stub loaderheight="4" tag="div" flat="true" class="content-scroll">
    <v-fade-transition-stub origin="top center 0">
      <nodata-stub message="We Couldn't find the Pokemon that you're looking for"></nodata-stub>
    </v-fade-transition-stub>
    <!---->
  </v-card-stub>
</v-card-stub>
`;

exports[`Pokemons List view renders error if query fails 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" width="100%" flat="true">
  <div class="overflow-hidden">
    <v-row-stub tag="div" nogutters="true">
      <v-col-stub cols="6" sm="6" md="6" tag="div">
        <v-btn-stub color="primary" elevation="0" tile="true" tag="button" activeclass="" large="true" block="true" type="button" class="text-capitalize">
          All
        </v-btn-stub>
      </v-col-stub>
      <v-col-stub cols="6" sm="6" md="6" tag="div">
        <v-btn-stub color="primary" elevation="0" tile="true" outlined="true" tag="button" activeclass="" large="true" block="true" type="button" class="text-capitalize">
          Favorites
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" align="center" justify="space-around">
      <v-col-stub cols="12" sm="6" md="7" tag="div" class="pr-3">
        <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" hidedetails="true" label="Search" loaderheight="2" clearable="true" clearicon="$clear" outlined="true" type="text"></v-text-field-stub>
      </v-col-stub>
      <v-col-stub cols="8" sm="4" md="4" tag="div">
        <v-select-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" appendicon="$dropdown" backgroundcolor="" hidedetails="true" label="Type" loaderheight="2" clearable="true" clearicon="$clear" outlined="true" type="text" valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date &amp;&amp; b instanceof Date &amp;&amp; a.getTime() !== b.getTime()) {
    // If the values are Date, compare them as timestamps
    return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}" nodatatext="$vuetify.noDataText" items="Grass,Poison,Fire,Flying,Water,Bug,Normal,Electric,Ground,Fairy,Fighting,Psychic,Rock,Steel,Ice,Ghost,Dragon" itemcolor="primary" itemdisabled="disabled" itemtext="text" itemvalue="value" menuprops="[object Object]" class="selectable"></v-select-stub>
      </v-col-stub>
      <v-col-stub cols="4" sm="2" md="1" tag="div">
        <v-row-stub tag="div" nogutters="true" align="center" justify="space-around">
          <v-btn-stub color="primary" tag="button" activeclass="" large="true" icon="true" type="button">
            <v-icon-stub large="">mdi-view-list</v-icon-stub>
          </v-btn-stub>
          <v-divider-stub vertical="true" align="center"></v-divider-stub>
          <v-btn-stub color="primary" tag="button" activeclass="" large="true" icon="true" type="button">
            <v-icon-stub large="">mdi-grid</v-icon-stub>
          </v-btn-stub>
        </v-row-stub>
      </v-col-stub>
    </v-row-stub>
  </div>
  <v-divider-stub class="elevation-10"></v-divider-stub>
  <v-card-stub loaderheight="4" tag="div" flat="true" class="content-scroll">
    <!---->
    <v-fade-transition-stub origin="top center 0">
      <nodata-stub message="We're down for maintenance. Check back shortly."></nodata-stub>
    </v-fade-transition-stub>
  </v-card-stub>
</v-card-stub>
`;
